pkg/gomega/assertions/eventually_csv.go

```diff
@@ -0,0 +1,41 @@
+package assertions
+
+import (
+	"context"
+	"log"
+
+	"github.com/onsi/gomega"
+	operatorv1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+// EventuallyCsv returns true if given namespace contains given CSV spec display name, and the CSV status is "Succeeded"
+// It can be used with the standard or custom gomega matchers, timeout and polling interval
+//
+//	EventuallyCsv(ctx, dynamicClient, operatorName, namespaceName).Should(BeTrue())
+func EventuallyCsv(ctx context.Context, dynamicClient dynamic.DynamicClient, specDisplayName, namespace string) gomega.AsyncAssertion {
+	return gomega.Eventually(func() bool {
+		csvList, err := dynamicClient.Resource(
```

Comment:
I think I'd prefer if we constructed the dynamic client in this function rather than requiring it be passed in. That way we have a common interface for interacting with these.